app/src/util.cc

```diff
@@ -77,32 +79,47 @@ static void PerformInitialize(ModuleInitializerData* data) {
 
 #if FIREBASE_PLATFORM_ANDROID
     if (init_result == kInitResultFailedMissingDependency) {
-      // On Android, we need to update or activate Google Play services
-      // before we can initialize this Firebase module.
-      LogWarning("Google Play services unavailable, trying to fix.");
-
-      Future<void> make_available = google_play_services::MakeAvailable(
-          data->app->GetJNIEnv(), data->app->activity());
-
-      make_available.OnCompletion(
-          [](const Future<void>& result, void* ptr) {
-            ModuleInitializerData* data =
-                reinterpret_cast<ModuleInitializerData*>(ptr);
-            if (result.status() == kFutureStatusComplete) {
-              if (result.error() == 0) {
-                LogInfo("Google Play services now available, continuing.");
-                PerformInitialize(data);
-              } else {
-                LogError("Google Play services still unavailable.");
-                int num_remaining = data->init_fns.size() - data->init_fn_idx;
-                data->future_impl.Complete(
-                    data->future_handle_init, num_remaining,
-                    "Unable to initialize due to missing Google Play services "
-                    "dependency.");
+      if (google_play_services::Initialize(data->app->GetJNIEnv(),
```

Comment:
Should this be checking g_initialized_google_play_services?  It seems odd to be Initializing here, especially since nothing cleans up after this call, unlike the other one.

Reply:
Added a comment to explain that it's Terminated in the OnCompletion below.